design/one-pager-provider-nop..md

```diff
@@ -0,0 +1,91 @@
+- be ready after this period
+- be unhealthy after this period
+
+The idea is to have an array of fields which will let you declare condition type and status of the resource at each time interval. 
+
+This will be implemented by making use of three fields:
+- `conditionType` and `conditionStatus` for declaring condtion of resource
+- `timeAfter` for declaring the time elapsed after the creation of resource at which we need to set the specified condition.
+
+The resource structure might look something like: 
+```go
+type ResourceConditionAfter struct {
+  Time            string `json:"time"`
+  ConditionType   string `json:"conditionType"`
+  ConditionStatus string `json:"conditionStatus"`
+}
+
+type NopResourceParameters struct {
+	ConditionAfter []ResourceConditionAfter `json:"conditionAfter"`
+}
+```
+
+### `NopResource` Controller
+
+The controller for this nop type will not call any external APIs during its reconcile loop and just perform the behaviour as specified by the user in the spec. 
+
+#### Setup
+Since default is too long for the testing use case, the sync period should be decreased to a small value like `100ms` using `WithPollInterval`in the Setup function. This will allow controller to set various conditions.
+
+#### Connect
+Currently, there is no requirement for having `ProviderConfig` since we don't have any external API client that we might want to connect. 
+We would just create an external client with no service parameter passed. 
+
+#### Observe
+This is where the main logic will reside that would compare the time elapsed with the time interval passed in the spec. As the time elapsed becomes equal to the time passed it will change the condition of resource accordingly. 
+
+#### Update
+Currently there is no plan to update the NopResource so this function will just set condition to crossplane-runtime `Available`. 
+
+#### Delete
+This will set the condition of NopResource to crossplane-runtime `Deleting`. 
+
+
+The config might look something like:
+
+```yaml
+apiVersion: sample.template.crossplane.io/v1alpha1
+kind: NopResource
+metadata:
+  name: example
+spec:
+  forProvider:
+    conditionAfter:
+      - conditionType: "Synced"
+        conditionStatus: "True"
+        timeAfter: "5s"
+      - conditionType: "Ready"
+        conditionStatus: "False"
+        timeAfter: "10s"
+```
+`ObservableField` is an arbitrary field in `NopResourceObservation` type can be useful in crossplane testing scenarios. For example, while testing bidirectional patching back to the composite resource from one resource to another which can be supported by this field. 
+
+The idea here is just to wait for the time provided in the spec before allowing the resource to be ready/unhealthy. This could be achieved by adding the logic to the controllers. 
+
+## Future Plans
+- We might want to include `ProviderConfig` at some point for those resources which are not managed. 
```

Comment:
can you explain this thought a little more? what resources are not managed?

Reply:
Yeah, I realize I might have put it in the wrong way. I was trying to point out that there might be an edge case where we want to test with `ProviderConfig`. I guess @muvaf was mentioning something about it that I might have misinterpreted.